src/pymor/vectorarrays/interfaces.py

```diff
@@ -82,6 +82,20 @@ def make_array(cls, subtype=None, count=0, reserve=0):
         """
         pass
 
+    @abstractclassmethod
+    def from_data(cls, data, subtype):
```

Comment:
Since `from_data` is really optional, I would prefer to not make it an abstract interace method but simply implement it in the interface with `raise NoteImplementedError`.